diff --git a/misc/scripts/download.sh b/misc/scripts/download.sh
index d84a789..42897b1 100755
--- a/misc/scripts/download.sh
+++ b/misc/scripts/download.sh
@@ -24,25 +24,34 @@
 
 # This script downloads pacscripts from the interwebs
 
-if curl --output /dev/null --silent --head --fail -- "$URL" ; then
+error_code="$(curl -o /dev/null -s -w "%{http_code}\n" -- "${URL}")"
+if [[ "${error_code}" != "200" ]]; then
+	if [[ "${error_code}" == "000" ]]; then
+		fancy_message error "Failed to download file, check your internet connection"
+		exit 1
+	fi
+	fancy_message error "Package does not exist"
+	exit 1
+fi
+if [[ "${error_code}" == "200" ]]; then
 	if [[ "$type" = "install" ]]; then
-		mkdir -p "$SRCDIR"
-		if ! cd "$SRCDIR" ; then
-			error_log 1 "install $PACKAGE"; fancy_message error "Could not enter ${SRCDIR}"; exit 1
+		mkdir -p "${SRCDIR}"
+		if ! cd "${SRCDIR}"; then
+			error_log 1 "install ${PACKAGE}"; fancy_message error "Could not enter ${SRCDIR}"; exit 1
 		fi
 	fi
 	
-	case "$URL" in
+	case "${URL}" in
 		*.pacscript)
-			wget -q --show-progress --progress=bar:force -- "$URL" > /dev/null 2>&1
+			wget -q --show-progress --progress=bar:force -- "${URL}" > /dev/null 2>&1
 		;;
 		*)
-			download -- "$URL" > /dev/null 2>&1
+			download -- "${URL}" > /dev/null 2>&1
 		;;
 	esac
 	return 0
 else
-	error_log 1 "get $PACKAGE pacscript"
+	error_log 1 "get ${PACKAGE} pacscript"
 	return 1
 fi
 # vim:set ft=sh ts=4 sw=4 noet:
